test/library/apiserver/apiserver.go

```diff
@@ -0,0 +1,123 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// WaitForAPIServerToStabilizeOnTheSameRevision waits until all API Servers are running at the same revision.
+// The API Servers must stay on the same revision for at least waitForAPIRevisionSuccessThreshold * waitForAPIRevisionSuccessInterval.
+// Mainly because of the difference between the propagation time of triggering a new release and the actual roll-out.
+//
+// Observations:
+//  rolling out a new version is not instant you need to account for a propagation time (~1/2 minutes)
+//  for some API servers (KAS) rolling out a new version can take ~10 minutes
+func WaitForAPIServerToStabilizeOnTheSameRevision(
+	t *testing.T,
+	podClient corev1client.PodInterface,
+	waitForAPIRevisionSuccessThreshold int,
```

Comment:
do we need all these parameters in every test? Isn't there a good default?

Reply:
sure we could try with what we have atm. Will work for KAS and OAS

```
waitForAPIRevisionSuccessThreshold = 3
waitForAPIRevisionSuccessInterval  = 1 * time.Minute

waitForAPIRevisionPollInterval = 30 * time.Second
waitForAPIRevisionTimeout      = 15 * time.Minute
```